scale/batch/messages/create_batch_recipes.py

```diff
@@ -91,6 +103,80 @@ def execute(self):
 
         self.new_messages.extend(new_messages)
         return True
+        
+    def _handle_new_batch(self, batch, definition):
+        """Handles creating a new batch of recipes with the defined dataset, returning any messages needed for the batch
+        
+        :param batch: The batch
+        :type batch: :class:`batch.models.Batch`
+        :param definition: The batch definition
+        :type definition: :class:`batch.definition.definition.BatchDefinition`
+        :return: The messages needed for the re-processing
+        :rtype: list
+        """
+        
+        messages = []
+
+        dataset = DataSet.objects.get(pk=definition.dataset)
+        dataset_definition = dataset.get_definition()
+        recipe_type = RecipeType.objects.get(pk=batch.recipe_type_id)
+        recipe_inputs = recipe_type.get_definition().get_input_keys()
+        
+        # No recipe inputs match the dataset 
+        if not any(elem in recipe_inputs for elem in dataset_definition.param_names):
+            return messages
```

Comment:
log error here? we should never have a batch defined where dataset parameters don't match the recipe inputs